multiversx_sdk_cli/cli_shared.py

```diff
@@ -154,13 +163,21 @@ def check_broadcast_args(args: Any):
         raise errors.BadUsage("Cannot both 'simulate' and 'send' a transaction")
 
 
+def check_guardian_and_options_args(args: Any):
+    check_guardian_args(args)
+    if args.guardian:
+        check_options_for_guarded_tx(args.options)
+
+
 def check_guardian_args(args: Any):
     if any([args.guardian, args.guardian_service_url, args.guardian_2fa_code]):
         if not all([args.guardian, args.guardian_service_url, args.guardian_2fa_code]):
             raise errors.BadUsage("All guardian arguments must be provided")
 
-        if not args.options & TRANSACTION_OPTIONS_TX_GUARDED == TRANSACTION_OPTIONS_TX_GUARDED:
-            raise errors.BadUsage("For guarded transactions the guarded flag must be set.")
+
+def check_options_for_guarded_tx(options: int):
+    if not options & TRANSACTION_OPTIONS_TX_GUARDED == TRANSACTION_OPTIONS_TX_GUARDED:
+        raise errors.BadUsage("For guarded transactions the 'guarded' flag must be set.")
```

Comment:
maybe refer to the `options` field. A better error would be "Invalid guarded transaction's options. The second least significant bit must be set"

Reply:
Done